### ISSUE TITLE
Misleading remote devices sync status during index exchange

### ISSUE THREAD

**Issue by scienmind** [syncthing/syncthing#4788]
Checking the behavior that'd be present at each version update once https://github.com/syncthing/syncthing/commit/d4b7be009c3cbae61b67a0306ff9f0d9d845bfe3 is in release, I noticed the following:

1. Run ST with "-reset-deltas" to simulate post-update run, initiating a full index exchange
2. Observe "Out of Sync Items" field for remote devices

Actual result: All shared items are listed as out of sync.
The syncing percentage and remaining transfer size also accounts for all shared files.

Expected result: List the actual "out of sync items" only once index exchange is complete, until then show "Updating Index" or something in that style. Percentage and size probably should also reflect the index exchange and not falsely claim that terabytes are about to be synced before the index is exchanged.

With delta indexes, the exchange is fast and this issue isn't noticeable, but a full index exchange for big nodes can be big (several GBs) and can take a while (hours), in the meantime showing to users the all the terabytes are out of sync. Which is misleading.

### Version Information

Syncthing Version: v0.14.45
OS Version: Windows 10x64

**Comment by calmh** [comment]
There is no protocol wise difference between a device not yet having synced a file and them not yet having sent us the index entry for it. We can not see the difference.

**Comment by scienmind** [comment]
So there is no way of differentiating "initial index exchange" from "just some index update for a new file that was just added"?

**Comment by calmh** [comment]
Currently no. There could be a protocol message that says “index complete, that’s all I have to send”, and we could then show “unknown” instead of out of sync until we receive that. But currently there isn’t.

**Comment by scienmind** [comment]
> There could be a protocol message that says “index complete, that’s all I have to send”, and we could then show “unknown” instead of out of sync until we receive that.

Yes, that's the basic idea: showing "updating index" or similar until it's known what the other side currently has.
